### ISSUE TITLE
test_classes_identities fails on 32-bit architectures

### ISSUE THREAD

**Issue by cjwatson** [psycopg/psycopg#945]
I noticed that `test_classes_identities` fails on Debian's 32-bit architectures:

- https://buildd.debian.org/status/fetch.php?pkg=psycopg3&arch=armel&ver=3.2.3-2&stamp=1730124139&raw=0
- https://buildd.debian.org/status/fetch.php?pkg=psycopg3&arch=armhf&ver=3.2.3-2&stamp=1730127211&raw=0
- https://buildd.debian.org/status/fetch.php?pkg=psycopg3&arch=i386&ver=3.2.3-2&stamp=1730122302&raw=0
- https://buildd.debian.org/status/fetch.php?pkg=psycopg3&arch=powerpc&ver=3.2.3-2&stamp=1730122744&raw=0

```
    def test_classes_identities():
        # Check if we know the class identities correctly. Maybe on different
        # platforms they are different.
        assert np.ubyte is np.uint8
        assert np.ushort is np.uint16
>       assert np.uint is np.uint64
E       AssertionError: assert <class 'numpy.uint32'> is <class 'numpy.uint64'>
E        +  where <class 'numpy.uint32'> = np.uint
E        +  and   <class 'numpy.uint64'> = np.uint64
```

I considered fixing this, but I can't work out what the point of this test is.  None of the left-hand-side types are used anywhere else in psycopg, and when I tried to trace the history of these tests in git I found that those types were _previously_ used in tests but aren't used any more.  Would the simplest fix for this be to just delete this test?

**Comment by dvarrazzo** [comment]
The point of this test is exactly this: to figure out if there are platforms on which `np.uint` is not 64 bits and other assumptions like that, and make sure that we don't register adapters of the wrong sizes.

I don't believe we build psycopg on 32 bits platform in our CI for distribution on PyPI. Is this the only test failing?

**Comment by cjwatson** [comment]
Yes, it seems to be the only test failing.  (I assume that some more of the assertions further down the test would fail, but haven't checked; I can put together test systems and give it a try if you need that information.)

**Comment by stankudrow** [comment]
> Yes, it seems to be the only test failing. (I assume that some more of the assertions further down the test would fail, but haven't checked; I can put together test systems and give it a try if you need that information.)

No need to, but you could help in reviewing this PR #946 .

**Comment by dvarrazzo** [comment]
Closed by #946

**Comment by cjwatson** [comment]
This still fails on 32-bit systems.  Fresh logs:

* https://buildd.debian.org/status/fetch.php?pkg=psycopg3&arch=armel&ver=3.2.3-3&stamp=1731264015&raw=0
* https://buildd.debian.org/status/fetch.php?pkg=psycopg3&arch=armhf&ver=3.2.3-3&stamp=1731263585&raw=0
* https://buildd.debian.org/status/fetch.php?pkg=psycopg3&arch=i386&ver=3.2.3-3&stamp=1731263490&raw=0
* https://buildd.debian.org/status/fetch.php?pkg=psycopg3&arch=powerpc&ver=3.2.3-3&stamp=1731263847&raw=0

```
    def test_classes_identities():
        # Check if we know the class identities correctly. Maybe on different
        # platforms they are different.
        size = _get_arch_size()
    
        assert np.byte is np.int8
        assert np.ubyte is np.uint8
    
        assert np.short is np.int16
        assert np.ushort is np.uint16
    
        if size == 32:
            assert np.uint is np.uint32
>           assert np.uintc is np.uint32
E           AssertionError: assert <class 'numpy.uintc'> is <class 'numpy.uint32'>
E            +  where <class 'numpy.uintc'> = np.uintc
E            +  and   <class 'numpy.uint32'> = np.uint32
```

I've done some digging and will propose another PR.
